### PR TITLE
refactor: implement various updates found during the Nx 14 migration

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**
- Some of our async tests that were using the deprecated `async()` function were timing out.
- Fixed the modals strict mode implementation.
- Various changes.

### CODE REVIEW

File: libs/components/packages/src/schematics/ng-add/ng-add.schematic.spec.ts [hunk 1]
```diff
@@ -37,6 +37,21 @@ describe('ng-add.schematic', () => {
...
+    );
+
+    const updatedTree = await runSchematic();
+
+    expect(
+      updatedTree
+        .readContent('projects/my-lib/src/test.ts')
+        .match(/\(window as any\)\.global = window/)?.length
+    ).toEqual(1);
+  });
```
Comment (Blackbaud-SteveBrush) [inline libs/components/packages/src/schematics/ng-add/ng-add.schematic.spec.ts:53]: ☝️ Added for 100% code coverage.

File: libs/sdk/e2e-schematics/src/utils/ast-utils.spec.ts [hunk 2]
```diff
@@ -3,10 +3,10 @@ import {
   componentGenerator,
 } from '@nrwl/angular/generators';
 import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
+import * as ts from '@schematics/angular/third_party/github.com/Microsoft/TypeScript/lib/typescript';
```
Comment (Blackbaud-SteveBrush) [inline libs/sdk/e2e-schematics/src/utils/ast-utils.spec.ts:6]: Import the same version of TypeScript that Angular uses.

File: libs/sdk/testing/src/lib/matchers/matchers.ts [hunk 3]
```diff
@@ -335,8 +336,8 @@ const matchers: jasmine.CustomMatcherFactories = {
         const actual = el.textContent;
 
         getResourcesObservable(name)
-          .toPromise()
-          .then((message) => {
+          .pipe(take(1))
```
Comment (Blackbaud-SteveBrush) [inline libs/sdk/testing/src/lib/matchers/matchers.ts:339]: ☝️ Remove the needless `toPromise()` calls.

File: libs/sdk/testing/src/lib/test-utility/test-utility.spec.ts [hunk 4]
```diff
@@ -14,15 +14,15 @@ import { SkyAppTestUtility } from './test-utility';
...
+  selector: 'test-parent-cmp',
   template: `
     <test-cmp [attr.data-sky-id]="'my-id'"> My component. </test-cmp>
   `,
 })
 class TestParentComponent {}
 
 @Component({
-  selector: 'sky-test-cmp',
+  selector: 'test-cmp',
```
Comment (Blackbaud-SteveBrush) [inline libs/sdk/testing/src/lib/test-utility/test-utility.spec.ts:25]: ☝️ The previous incorrect selector was throwing console warnings.

**Review by Blackbaud-PaulCrowder — APPROVED** [review]

**Pull request merged**